pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -270,6 +271,33 @@ func (c *Cluster) startProvisioning(config *provisionConfig) {
 			}
 		}
 	}
+
+	// Deploying test based on the storageclassDeviceSet
+	// Todo: Replace the below logic to deploy osd pods over pvcs
+	for _, storageclassDeviceSet := range c.DesiredStorage.Storage {
+		for i := 0; i < storageclassDeviceSet.Count; i++ {
+			sc := "test-sc"
+			pvc := &v1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pvc-" + storageclassDeviceSet.Name + fmt.Sprintf("-%v", i),
+					Namespace: "rook-ceph",
+				},
+				Spec: v1.PersistentVolumeClaimSpec{
+					AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce, v1.ReadOnlyMany},
```

Comment:
For OSDs we need `volumeMode: Block`

Reply:
cool we do it

Reply:
it will be passed through the pvc spec from storageclassdeviceset.